tox.ini

```diff
@@ -24,6 +24,8 @@ format = pylint
 [testenv:syntax]
 deps = 
     flake8
+    # NOTE(retr0h): We specifically pin to a known version so formatting
+    # is consistent across developer/PR.
     yapf==0.10.0
```

Comment:
That's only one use case. For instance my editor needs know where yapf is, because it reformats my the files every time I save them.

The other concern is totally valid. Here is a pattern I use often to solve that:
```
├── requirements
│   ├── _base.txt
│   ├── _docs.txt
│   ├── local.txt
│   ├── production.txt
│   └── _tests.txt
├── requirements.txt
```
Where:

1. local.txt installs the base, docs and tests packages 
2. production installs base only
3. The main requirements.txt always installs production